src/upload.test.ts

```diff
@@ -256,3 +257,84 @@ describe("uploadDirectory", () => {
     );
   });
 });
+
+describe("splitSizeIntoChunkAlignedParts", () => {
+  const mib = 1 << 20;
+  const sizesAndChunks: Array<[number, number, { start: number; end: number }[]]> = [
+    [
+      40 * mib,
+      2,
```

Comment:
What happens if you do 3 chunks and 40 mib? Will it have 1 full chunk and 2 empty ones?

Reply:
Yup, I just confirmed it with this test case:

```ts
[
  40 * mib,
  3,
  [
    { start: 0, end: 40 * mib },
    { start: 40 * mib, end: 40 * mib },
    { start: 40 * mib, end: 40 * mib },
  ],
],
```

I won't push it here since it's already approved, but I'll add this case in a followup PR.